lib/Core/Json.ts

```diff
@@ -16,6 +16,15 @@ export function isJsonObject(
   );
 }
 
+export function isJsonArray(value: JsonValue | undefined): value is JsonArray {
+  return (
+    value !== undefined &&
+    value !== null &&
+    Array.isArray(value) &&
+    value.every(child => isJsonObject(child) || isJsonArray(child))
+  );
+}
+
```

Comment:
~This will return false for `isJsonArray(['a'])` and `['a']` is otherwise a valid `JsonArray`. This function needs to be renamed to avoid this confusion.~
Moved to ShapefileCatalogItem and renamed to `isJsonArrayOrDeepArrayOfObjects`. I don't object to this function being in `Json.ts` if it's more widely applicable, but using this function instead of `Array.isArray` could cause errors/confusion and could be much slower given it's recursive nature.